README.md
# groovin_material_icons

This Flutter package is a fork of community_material_icons, which seems to have been abandoned. There are some icons that the original developer added to the repo, but was never released on pub. This package includes those icons.

If you find any icons that are not in the package and should be, please either open an issue linking to the icon or fork the repo, add the icon yourself, and open a pull request

## Special thanks to:
- Miyoyo
- ThinkDigital
- Matthew Evans

## How to contribue
1. Fork the repo
2. Clone the repo to your machine
3. Add your new IconData. Make sure that you know the codePoint for your icon. Check https://cdn.materialdesignicons.com/2.6.95/ for your icon if it is an icon that the community has already created. If you have an svg icon, you can upload it to http://fluttericon.com/ to generate a custom font that will tell you exactly how to add your icon to the package.
4. Commit your code to your repo
5. Open a pull request

## Getting Started with Flutter

For help getting started with Flutter, view our online [documentation](https://flutter.io/).

For help on editing package code, view the [documentation](https://flutter.io/developing-packages/).




CHANGELOG.md
## 1.1.0 - 9/13/2018
* Version 1.1.6 updates incorrect dependencies

* Version 1.1.5 updates the 'url_launcher' dependency

* Version 1.1.4 updates the example app

* Version 1.1.3 updates the example app with icon search and a new view when tappping an icon

* Version 1.1.2 updates the example app with all icons included in the package

* Version 1.1.1 adds the send_outline icon

* Version 1.1.0 starts to introduce Google's new outline icons. Look out for more coming soon!

## 1.0.0 - 8/15/2018

* Version 1.0.6 introduces the dart icon to the package

* Version 1.0.5 fixes flutter sdk in pubspec

* Version 1.0.4 adds the Flutter icon to the package

* Version 1.0.3 fixed Flutter version

* Version 1.0.2 fixed example

* Version 1.0.1 removes the word 'community' from the class name

* Version 1.0.0 brings new material icons into the package

example/test/widget_test.dart
// This is a basic Flutter widget test.
// To perform an interaction with a widget in your test, use the WidgetTester utility that Flutter
// provides. For example, you can send tap and scroll gestures. You can also use WidgetTester to
// find child widgets in the widget tree, read text, and verify that the values of widget properties
// are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:groovin_material_icons_testapp/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {

  });
}


test/groovin_material_icons_test.dart
/*
import 'package:test/test.dart';

import 'package:groovin_material_icons/groovin_material_icons.dart';

void main() {
  test('adds one to input values', () {

  });
}
*/


example/lib/search.dart
import 'package:flutter/material.dart';
import 'package:groovin_material_icons/groovin_material_icons.dart';
import 'package:groovin_material_icons_testapp/icon_map.dart';

import 'icon_display.dart';

class IconSearch extends SearchDelegate<String> {
  List<String> recentSearches = [];

  ///a [List] of Icons, usually, that will show up at the end of the search bar.
  @override
  List<Widget> buildActions(BuildContext context) {
    return [
      IconButton(
        icon: Icon(GroovinMaterialIcons.close),
        onPressed: () {
          query = "";
        },
      )
    ];
  }

  ///a [List] of Icons, usually, that will show up at the beginning of the search bar.
  @override
  Widget buildLeading(BuildContext context) {
    return IconButton(
      icon: AnimatedIcon(
        icon: AnimatedIcons.menu_arrow, progress: transitionAnimation,
      ),
      onPressed: () {
        close(context, null);
      },
    );
  }

  @override
  Widget buildResults(BuildContext context) {
    return Center(
      child: IconDisplay(
        title: query,
        iconData: iconMap[query],
      ),
    );
  }

  @override
  Widget buildSuggestions(BuildContext context) {
    final suggestionList =
        iconMap.keys.where((iconName) => iconName.contains(query)).toList();
    return ListView.builder(
      itemCount: suggestionList.length,
      itemBuilder: (BuildContext context, int index) {
        return ListTile(
          leading: iconMap.containsKey(suggestionList[index])
              ? Icon(iconMap[suggestionList[index]])
              : Icon(GroovinMaterialIcons.border_none),
          title: Text(suggestionList[index]),
          onTap: () {
            query = suggestionList[index];
            showResults(context);
          },
        );
      },
    );
  }
}


example/lib/icon_display.dart
import 'package:flutter/material.dart';

class IconDisplay extends StatelessWidget {
  final IconData iconData;
  final String title;
  final bool showAppBar;

  IconDisplay(
      {@required this.title, @required this.iconData, this.showAppBar = false});

  final List<Color> colorList = [
    Colors.black,
    Colors.blue,
    Colors.red,
    Colors.yellow,
    Colors.green
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: showAppBar
        ? AppBar(
          centerTitle: false,
          iconTheme: IconThemeData(color: Colors.black),
          backgroundColor: Colors.white,
          toolbarOpacity: 1.0,
          title: Text(
            title,
            style: TextStyle(fontSize: 20.0, color: Colors.black),
          ),
        )
        : null,
      body: Card(
        child: Padding(
          padding: const EdgeInsets.symmetric(vertical: 10.0),
          child: Center(
            child: ListView.builder(
              physics: ClampingScrollPhysics(),
              shrinkWrap: true,
              itemCount: colorList.length,
              itemBuilder: (BuildContext context, int index) {
                double iconSize = (index + 1) * 35.toDouble();
                return Padding(
                  padding: const EdgeInsets.symmetric(horizontal: 8.0),
                  child: Icon(
                    iconData,
                    color: colorList[index],
                    size: iconSize,
                  ),
                );
              },
            ),
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:draggable_scrollbar/draggable_scrollbar.dart';
import 'dart:async';
import 'package:flutter/material.dart';
import 'package:groovin_material_icons/groovin_material_icons.dart';
import 'package:groovin_material_icons_testapp/icon_map.dart';
import 'package:groovin_material_icons_testapp/search.dart';
import 'package:url_launcher/url_launcher.dart';

import 'icon_display.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'GMI Companion',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'GMI Companion'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  var scrollController = ScrollController();
  var fabIcon = Icons.search;
  Function refreshFab;
  var iconList;

  @override
  void initState() {
    iconList = iconMap
      .map((title, icon) => MapEntry(
      title,
      new IconCard(
        title: title,
        icon: icon,
      )))
      .values
      .toList();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    var crossAxisCount = (MediaQuery.of(context).size.width / 150).ceil();
    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        centerTitle: true,
        backgroundColor: Colors.white,
        iconTheme: IconThemeData(
          color: Colors.black
        ),
        title: Text(widget.title, style: TextStyle(color: Colors.black)),
        leading: IconButton(
          icon: Icon(Icons.info_outline),
          onPressed: (){
            showModalBottomSheet(
              context: context,
              builder: (builder){
                return Container(
                  child: SingleChildScrollView(
                    child: Column(
                      mainAxisSize: MainAxisSize.min,
                      children: <Widget>[
                        Padding(
                          padding: const EdgeInsets.only(top: 8.0),
                          child: Row(
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: <Widget>[
                              Container(
                                height: 5.0,
                                width: 25.0,
                                decoration: BoxDecoration(
                                    color: Colors.grey[300],
                                    borderRadius: BorderRadius.only(
                                      topLeft: const Radius.circular(10.0),
                                      topRight: const Radius.circular(10.0),
                                      bottomLeft: const Radius.circular(10.0),
                                      bottomRight: const Radius.circular(10.0),
                                    )
                                ),
                              ),
                            ],
                          ),
                        ),
                        ListTile(
                          leading: Icon(GroovinMaterialIcons.information_variant, color: Colors.black,),
                          title: Text("Package Version: 1.1.4"),
                        ),
                        Material(
                          child: ListTile(
                            leading: Icon(GroovinMaterialIcons.dart_logo, color: Colors.blue,),
                            title: Text("View on Pub"),
                            onTap: (){
                              launch("https://pub.dartlang.org/packages/groovin_material_icons");
                            },
                          ),
                        ),
                        ListTile(
                          leading: Icon(GroovinMaterialIcons.creation, color: Colors.black,),
                          title: Text("Special Thanks to:"),
                          subtitle: Text("Miyoyo, Matthew Evans, and ThinkDigital"),
                        ),
                        Divider(
                          color: Colors.grey,
                          height: 0.0,
                        ),
                        ListTile(
                          leading: Icon(GroovinMaterialIcons.flutter, color: Colors.blue),
                          title: Text("Get Started with Flutter"),
                          onTap: (){
                            launch("https:flutter.io");
                          },
                        ),
                        Divider(
                          color: Colors.grey,
                          height: 0.0,
                        ),
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                          children: <Widget>[
                            IconButton(
                              icon: Icon(GroovinMaterialIcons.twitter, color: Colors.blue),
                              onPressed: (){
                                launch("https:twitter.com/GroovinChipDev");
                              },
                            ),
                            IconButton(
                              icon: Icon(GroovinMaterialIcons.github_circle),
                              onPressed: (){
                                launch("https:github.com/GroovinChip");
                              },
                            ),
                            IconButton(
                              icon: Icon(GroovinMaterialIcons.gmail),
                              color: Colors.red,
                              onPressed: (){
                                launch("mailto:groovinchip@gmail.com");
                              },
                            ),
                            IconButton(
                              icon: Icon(GroovinMaterialIcons.discord, color: Colors.deepPurpleAccent),
                              onPressed: (){
                                launch("https://discord.gg/CFnBRue");
                              },
                            ),
                          ],
                        ),
                      ],
                    ),
                  ),
                );
              }
            );
          },
        ),
        actions: <Widget>[
          IconButton(
            icon: Icon(Icons.search),
            onPressed: () {
              showSearch(context: context, delegate: IconSearch());
            },
          ),
        ],
      ),
      body: DraggableScrollbar.rrect(
        backgroundColor: Colors.indigo,
        controller: scrollController,
        child: GridView.builder(
          physics: PageScrollPhysics(),
          controller: scrollController,
          padding: EdgeInsets.all(10.0),
          itemCount: iconMap.length,
          gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
              crossAxisCount: crossAxisCount),
          itemBuilder: (BuildContext context, int index) {
            return iconList[index];
          },
        ),
      ),
    );
  }
}

class IconCard extends StatelessWidget {
  final String title;
  final IconData icon;

  const IconCard({
    Key key,
    this.title,
    this.icon,
  }) : super(key: key);

  final double elevation = 0.0;
  final double iconSize = 35.0;

  final FontWeight fontWeight = FontWeight.w300;

  @override
  Widget build(BuildContext context) {
    return Card(
      elevation: elevation,
      child: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            IconButton(
              icon: Icon(icon),
              iconSize: iconSize,
              onPressed: () {
                Navigator.of(context)
                    .push(MaterialPageRoute(builder: (BuildContext context) {
                  return IconDisplay(
                    title: title,
                    iconData: icon,
                    showAppBar: true,
                  );
                }));
              },
            ),
            Padding(
              padding: const EdgeInsets.all(8.0),
              child: Text(
                title,
                textAlign: TextAlign.center,
                style: TextStyle(fontWeight: fontWeight),
              ),
            )
          ],
        ),
      ),
    );
  }
}
